engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -410,13 +437,15 @@ public void writeTuple(Object obj, String port)
       storage.writeDataItem(bos.toByteArray(), true);
       //logger.debug("Writing tuple for port id {}", pi.id);
       //fsOutput.hflush();
-      ++totalTupleCount;
       if (numSubscribers > 0) {
         // this is not asynchronous.  we need to fix this
         publishTupleData(pi.id, obj);
       }
-    } catch (IOException ex) {
-      logger.error(ex.toString());
+      ++totalTupleCount;
```

Comment:
This is from existing code and not introduced for this fix and was counting only successfully processed tuples. It was moved so that the correct count is available when evaluating the logging during exception in publishTupleData. It can be changed to a long.